pkg/proc/redirect/redirect.go

```diff
@@ -0,0 +1,20 @@
+package redirect
+
+import (
+	"errors"
+	"os"
+)
+
+var ErrorNotImplemented = errors.New("not implemented")
+
+// Redirect
+type Redirect interface {
```

Comment:
Make this a concrete type, put it in pkg/proc, not in its own package, make it describe a single redirect rather than three redirects in one. Do not repeat the type name in the name of its methods. For example, it should be (*Redirect).Path not (*Redirect).RedirectPath.

Restart is a word, it shouldn't be spelled `ReStart`.

Reply:
done.

Reply:
> make it describe a single redirect rather than three redirects in one

Still TBD.

Reply:
Do you mean `WriterFiles [3]*os.File -> StdinWriter *os.File, StdoutWriter *os.File, StderrWriter *os.File`？

Reply:
I think this should just be a string/io.Writer pair:

```
type OutputRedirect struct {
     Path string
     Writer io.Writer
}
```

Reply:
How about this?
```
type OutputRedirect struct {
	Path string
	File *os.File
}

Launch(.....,redirect [3]OutputRedirect) 
```

I can't use `io.Writer` or `io.WriteColser`, because in `gdbserial/rr.go` it needs `*os.File`.
In `service/service.go`, the data will be read from this structure.

Reply:
Ok.

Reply:
@aarzilli  Need a reply.

Reply:
What you suggested is ok.